app/src/components/market/sections/market_pooling/market_pool_liquidity.tsx

```diff
@@ -782,7 +782,10 @@ const MarketPoolLiquidityWrapper: React.FC<Props> = (props: Props) => {
         />
       )}
       <BottomButtonWrapper borderTop>
-        <Button buttonType={ButtonType.secondaryLine} onClick={() => history.goBack()}>
+        <Button
+          buttonType={ButtonType.secondaryLine}
+          onClick={() => (history.length > 2 ? history.goBack() : history.replace('/liquidity'))}
```

Comment:
Looks good! Just not sure why did you include this if else statment? I would suggest just putting to redirect to /liquidity since that is desired behaviour...

Reply:
@pimato requested that feature as well. So if the user had some actif filter it goes back that seafch if not, it goes to the main market page

Reply:
ohh yeah...Works great!